### PR TITLE
RefreshIndicator should appear from the top of the dragged widget where the [GlowingOverscrollIndicator] appears position.

### LINKED ISSUE #71819: RefreshIndicator should appear from the top of the dragged widget where the [GlowingOverscrollIndicator] appears position.
## Steps to Reproduce
1. Run the sample code.
2. Pull out the `RefreshIndicator` by dragging it down **slowly**.
3. Observe where the indicator animation begins to appear.

The animated [RefreshIndicator] should appear from the top of the dragged widget
where the [GlowingOverscrollIndicator] (if have) appears position.

## Sample Code
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(home: BugReport(),));
}

class BugReport extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    double leadingPaintOffset = MediaQuery.of(context).padding.top + AppBar().preferredSize.height;
    return RefreshIndicator(
      onRefresh: () {
        return Future.delayed(const Duration(seconds: 3), () => print('onRefresh finished.'));
      },
      child: NotificationListener<OverscrollIndicatorNotification>(
        onNotification: (notification) {
          if (notification.leading) {
            notification.paintOffset = leadingPaintOffset;
          }
          return false;
        },
        child: CustomScrollView(
          slivers: [
            SliverAppBar(title: Text('Custom PaintOffset')),
            SliverToBoxAdapter(
              child: Container(
                color: Colors.amberAccent,
                height: 100,
                child: Center(child: Text('I love Flutter!', style: TextStyle(fontSize: 18.0),)),
              ),
            ),
            SliverFillRemaining(child: FlutterLogo()),
          ],
        ),
      ),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by xu-baolin**
## Description

This change fixes that RefreshIndicator does not appear from the top of the dragged widget where the [GlowingOverscrollIndicator] appears position when custom the position of GlowingOverscrollIndicator.
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(home: BugReport(),));
}

class BugReport extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    double leadingPaintOffset = MediaQuery.of(context).padding.top + AppBar().preferredSize.height;
    return RefreshIndicator(
      onRefresh: () {
        return Future.delayed(const Duration(seconds: 3), () => print('onRefresh finished.'));
      },
      child: NotificationListener<OverscrollIndicatorNotification>(
        onNotification: (notification) {
          if (notification.leading) {
            notification.paintOffset = leadingPaintOffset;
          }
          return false;
        },
        child: CustomScrollView(
          slivers: [
            SliverAppBar(title: Text('Custom PaintOffset')),
            SliverToBoxAdapter(
              child: Container(
                color: Colors.amberAccent,
                height: 100,
                child: Center(child: Text('I love Flutter!', style: TextStyle(fontSize: 18.0),)),
              ),
            ),
            SliverFillRemaining(child: FlutterLogo()),
          ],
        ),
      ),
    );
  }
}
```

## Related Issues

Fixes #71819

## Tests

I added the following tests:

See files.

### CODE REVIEW

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 1]
```diff
@@ -56,6 +56,42 @@ enum _RefreshIndicatorMode {
 /// scrollable's contents and then complete the [Future] it returns. The refresh
 /// indicator disappears after the callback's [Future] has completed.
 ///
+/// The animated [RefreshIndicator] will appear from the top of the dragged widget
+/// where the [GlowingOverscrollIndicator] (if have) appears position. If you provide a
+/// [OverscrollIndicatorNotification.paintOffset] to custom the position of
+/// GlowingOverscrollIndicator, you should insert the [NotificationListener] add between
+/// the [RefreshIndicator] and scrollable widgets, otherwise, the RefreshIndicator
+/// will cancel the notification bubbling:
```
Comment (xu-baolin) [inline packages/flutter/lib/src/material/refresh_indicator.dart:64]: should be `otherwise, the RefreshIndicator can not receive the notification:`

**Comment by xu-baolin** [comment]
@darrenaustin Hi,
Welcome to review me :)

**Comment by darrenaustin** [comment]
@xu-baolin, thanks for the contribution. Sorry it has taken me a while to look at this. I just looked at another related PR #72052 and it would appear that would solve this problem as well. Depending on how it lands, the app could just set the `displacement` (or `edgeDistance` if that is the decision for the PR) on the `RefreshIndicator` to match where the glow indicator is starting. Which I think would handle the same situation, but without the need for the RefreshIndicator to handle this case specifically.

Can you take a look at #72052 and see if it wouldn't solve the same issue by just updating your test program with:

```dart
    ...

    return RefreshIndicator(
      onRefresh: () {
        return Future.delayed(const Duration(seconds: 3), () => print('onRefresh finished.'));
      },
     displacement: leadingPaintOffset,

    ...
```

**Comment by xu-baolin** [comment]
Thanks for reviewing.
I just take a look at that PR,
I think the situation is a bit different. My understanding is that if you change the position of the glow, the position of the indicator should automatically follow, instead of letting the user manually set it, and ensure that the values on both sides are the same to work properly. This is unfriendly.

**Comment by xu-baolin** [comment]
The key is whether the indicator and glow should appear from the same position by default?
 If the answer is yes, should we need also to change the position of the glow when changing the position of the indicator like #72052 ?

**Comment by Piinks** [comment]
Hey @xu-baolin! I just came across this, would you still like to proceed with this change?

**Comment by xu-baolin** [comment]
I have no idea, #72052 seems to help this,if you think so,just close this:)

**Comment by Piinks** [comment]
Thank you! Much appreciated as always. :)

**Comment by darrenaustin** [comment]
Indeed. Thank you both. I have been meaning to follow up on this for a while and it got buried in my task list.

**Pull request closed without merging**